src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,277 @@
+            }
+
+            return elementToReturn.item;
+        }
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " does not exist");
+        }
+
+        Node<T> elementToSet = new Node<>(value);
+        Node<T> elementToReturn = null;
+
+        if (size == 1) {
+            elementToReturn = head;
+            head = elementToSet;
+            tail = elementToSet;
+        } else if (index == size - 1) {
+            elementToReturn = tail;
+            Node<T> elPrevTail = tail.prev;
+            elPrevTail.next = elementToSet;
+            elementToSet.prev = elPrevTail;
+            tail = elementToSet;
+        } else if (index == 0) {
+            elementToReturn = head;
+            Node<T> elNextHead = head.next;
+            elNextHead.prev = elementToSet;
+            elementToSet.next = elNextHead;
+            head = elementToSet;
+        } else {
+            int countOfIndex = 0;
+            Node<T> element = head;
+
+            while (countOfIndex != index) {
+                element = element.next;
+                countOfIndex++;
+            }
+
+            elementToReturn = element;
+
+            Node<T> elPrevElement = elementToReturn.prev;
+            Node<T> elNextElement = elementToReturn.next;
+
+            elPrevElement.next = elementToSet;
+            elementToSet.prev = elPrevElement;
+
+            elNextElement.prev = elementToSet;
+            elementToSet.next = elNextElement;
+        }
+
+        return elementToReturn.item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " does not exist");
+        }
+
+        Node<T> elementToRemove = null;
+
+        if (size == 1) {
+            elementToRemove = head;
+            head = null;
+            tail = null;
+        } else if (index == size - 1) {
+            elementToRemove = tail;
+            Node<T> elPrevElementToRemove = tail.prev;
+            elPrevElementToRemove.next = null;
+            tail = elPrevElementToRemove;
+        } else if (index == 0) {
+            elementToRemove = head;
+            Node<T> elNextElementToRemove = head.next;
+            elNextElementToRemove.prev = null;
+            head = elNextElementToRemove;
+        } else {
+            int countOfIndex = 0;
+            Node<T> element = head;
+
+            while (countOfIndex != index) {
+                element = element.next;
+                countOfIndex++;
+            }
+
+            elementToRemove = element;
+            Node<T> elPrevElementToRemove = elementToRemove.prev;
+            Node<T> elNextElementToRemove = elementToRemove.next;
+
+            elPrevElementToRemove.next = null;
+            elNextElementToRemove.prev = null;
```

Comment:
In the `remove` method, setting `next` and `prev` to `null` before updating the pointers to bypass the node is redundant and incorrect. This can lead to breaking the list structure.